### DISCUSSION TITLE
How do i refer to the current vm resource using self?

### DISCUSSION THREAD

**Discussion by squiros87** [bpg/terraform-provider-proxmox#1625] [Q&A]
hi im using the block proxmox_virtual_environment_vm and i need to refer the ip of the vm inside the block using the provisioner file can i try using self?.

Regards

**Comment by bpg-dev** [comment]
Hey @squiros87 👋🏼 

If you're using DHCP, the VM IP will be known only after `apply` not during the `plan` stage. So if you need to pass the actual IP value somewhere _inside_ the VM resource block, it won't be available.
But you can reference it outside of the VM resource.

I could help you with specifics if you provide more details on the problem you are trying to solve.
